### PR TITLE
Fix validation script after adding structured errors

### PR DESCRIPTION
**Pull request by amanusk**
This does requires messages to be a list, which might not be what we want

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/235)

### CODE REVIEW

**Comment by kkovaacs** [comment]
This is great -- but I don't think we should change the specification in that way just to make validation possible.

Looks like there's an open issue on `@json-schema-tools/dereferencer` for handling recursive schemas: https://github.com/json-schema-tools/dereferencer/issues/608

Would it be possible to:
- rebase this branch to the current state of things (after merging the `starknet_getStorageProof` change)
- include just the schema fixes that are purely fixes and maybe skip that single change that makes the child CONTRACT_EXECUTION_ERROR an array?

**Comment by amanusk** [comment]
Agree that it is a bit backwards and not ideal.
I'll try exploring ways to fix the validation script in a way that does not require this change.
But it would be a shame to forgo all validations going forward, and since this component is in the main API schema, it breaks validating all other schemas as well :(

File: api/starknet_api_openrpc.json [hunk 1]
```diff
@@ -3756,20 +3750,20 @@
...
         }
-    },
-    "CONTRACT_EXECUTION_ERROR": {
+      },
+      "CONTRACT_EXECUTION_ERROR": {
+        "description": "structured error that can later be processed by wallets or sdks",
+        "title": "contract execution error",
+        "$ref": "#/components/schemas/CONTRACT_EXECUTION_ERROR_INNER"
+      },
+      "CONTRACT_EXECUTION_ERROR_INNER": {
```
Comment (kkovaacs) [inline api/starknet_api_openrpc.json:3761]: Ah, this is a much neater trick than the previous one! Thanks for finding out how to make this work! :+1:

Comment (amanusk) [reply api/starknet_api_openrpc.json:3761]: Good thing you insisted

**Review by ArielElp — APPROVED** [review]
:lgtm:

> Reviewed 1 of 3 files at r1, 3 of 3 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/235#-:-OB5oJ9QBRAUz5_jZY8W:bnfp4nl)* status: all files reviewed (commit messages unreviewed), 1 unresolved discussion (waiting on @kkovaacs)

**Review by ArielElp — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/235)* status: all files reviewed, 1 unresolved discussion (waiting on @kkovaacs)

**Review by ArielElp — APPROVED** [review]
> Dismissed @kkovaacs from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/235)* status: all files reviewed, 1 unresolved discussion (waiting on @ArielElp and @kkovaacs)

**Review by ArielElp — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/235)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @amanusk)

**Pull request merged**